### PR TITLE
Refactor finding detail and fix finding search routes

### LINKED ISSUE #688: Finding routes don't work when component or vulnerability specified
### Current Behavior

There is existing functionality for showing a specific vulnerability in the UI, but it isn't advertised and doesn't work since #667.

### Steps to Reproduce

Browse to `/projects/{id}/findings/CVE-2023-24534`. It will open the audit vulnerabilities tab with a search for `:CVE-2023-24534` pre-populated.

Browse to `/projects/{id}/findings/{proj uuid}/{vuln uuid}`. It will populate the search with `{vuln uuid}` only, no component ID.

### Expected Behavior

The audit tab is opened with a search for either the specified vulnerability, or `component:vulnerability` to allow you to search for the matrix of a specific finding.

### Dependency-Track Frontend Version

4.11.0-SNAPSHOT

### Browser

Mozilla Firefox

### Browser Version

_No response_

### Operating System

macOS

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/frontend/issues) for whether this defect was already reported

### PR DESCRIPTION
**Pull request by mykter**
### Description

This PR fixes a pair of bugs and factors out the finding detail view into its own single file component.

It might be easiest to review commit by commit.

### Addressed Issue

Fixes #688 in 3895307e761d56e259b340edfc6e9a320db98e41

### Additional Details

I originally planned to make a new view to show a specific finding, and wanted to re-use the detail view from the audit page. This was inlined in a way that was very hard to re-use (or read!). In my journey to try and re-use it I came across [Steve's earlier investigations](https://github.com/wenzhixin/bootstrap-table/issues/4754), and was eventually able to pull it out into its own component as it looks like he originally wanted to.

I then discovered that there is [already a route](https://github.com/DependencyTrack/frontend/blob/finding-audit-route/src/router/index.js#L134) that shows this information, it just doesn't work.

I think the refactoring is still worthwhile by itself - maintaining this view in SFC form is easier than in the inline form. The refactoring allowed eslint to find some [issues](https://github.com/DependencyTrack/frontend/pull/689/commits/b10be4c10449d9c0b8b309032ee7699ff2bccd7c) in the inlined code, for example.

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#pull-requests)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

File: src/views/portfolio/projects/FindingAudit.vue [hunk 1]
```diff
@@ -0,0 +1,246 @@
+<template>
```
Comment (mykter) [inline src/views/portfolio/projects/FindingAudit.vue:1]: Aside from the imports and structure, this is all just a copy-and-paste from the inline version.

File: src/views/portfolio/projects/ProjectFindings.vue [hunk 2]
```diff
@@ -239,7 +242,7 @@ import common from "../../../shared/common";
           pageSize: (localStorage && localStorage.getItem("ProjectFindingsPageSize") !== null) ? Number(localStorage.getItem("ProjectFindingsPageSize")) : 10,
           sortName: (localStorage && localStorage.getItem("ProjectFindingsSortName") !== null) ? localStorage.getItem("ProjectFindingsSortName") : undefined,
           sortOrder: (localStorage && localStorage.getItem("ProjectFindingsSortOrder") !== null) ? localStorage.getItem("ProjectFindingsSortOrder") : undefined,
-          searchText: this.$route.params.vulnerability ? ":" + this.$route.params.vulnerability : undefined,
+          searchText: this.$route.params.vulnerability ? this.$route.params.vulnerability : undefined,
```
Comment (mykter) [inline src/views/portfolio/projects/ProjectFindings.vue:245]: This is the bug fix

File: vue.config.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ module.exports = {
     proxy: { "/api": { target: process.env.VUE_APP_SERVER_URL} }
   },
   configureWebpack: {
+    devtool: 'source-map',
```
Comment (mykter) [inline vue.config.js:14]: This is a drive-by improvement from [the docs](https://v2.vuejs.org/v2/cookbook/debugging-in-vscode.html#Displaying-Source-Code-in-the-Browser), which I used to make debugging easier.

File: src/router/index.js [hunk 4]
```diff
@@ -132,7 +132,7 @@ function configRoutes() {
         },
         {
           path: 'projects/:uuid/findings/:vulnerability',
-          name: 'Project Finding Lookup',
+          name: 'Project Vulnerability Lookup',
```
Comment (mykter) [inline src/router/index.js:135]: I noticed this existing name clash because vue-router was complaining about it in the console.

**Review by nscuro — APPROVED** [review]
Thanks @mykter, I *really* appreciate the attention to detail in this PR. 

From laying out your reasoning in the description, to making sure each change is understandable one-by-one, this was a pleasure to look at.

**Pull request merged**